CMakeLists.txt

```diff
@@ -830,6 +832,28 @@ else()
 	endif (MSVC)
 endif()
 
+if (io_uring)
+	find_public_dependency(liburing QUIET)
+	if (liburing_FOUND)
+		target_link_libraries(torrent-rasterbar PUBLIC liburing)
+	else()
+		include(FindPkgConfig)
+		pkg_check_modules(liburing REQUIRED IMPORTED_TARGET GLOBAL liburing)
+		target_link_libraries(torrent-rasterbar PUBLIC PkgConfig::liburing)
+
+		# propagate the dependency properly to help downstream project find liburing
+		set(_pkg_check_modules_calls
+			"find_dependency(PkgConfig)"
+			"pkg_check_modules(liburing REQUIRED QUIET IMPORTED_TARGET GLOBAL liburing)")
+		string(REPLACE ";" "\n" _pkg_check_modules_calls "${_pkg_check_modules_calls}")
+	endif()
+
+	target_compile_definitions(torrent-rasterbar PUBLIC
+		BOOST_ASIO_DISABLE_EPOLL
+		BOOST_ASIO_HAS_IO_URING
+		BOOST_ASIO_HAS_IO_URING_AS_DEFAULT)
```

Comment:
from the documentation "In general, BOOST_ASIO_HAS_ macros should not be explicitly defined by the user,"

https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/using.html

Reply:
Yeah, considering the above, the documentation simply contradicts itself.